src/bin/pg_autoctl/pgsql.c

```diff
@@ -1601,6 +1601,30 @@ pgsql_set_default_transaction_mode_read_write(PGSQL *pgsql)
 }
 
 
+/*
+ * pgsql_set_password sets the password for the given username.
+ */
+bool
+pgsql_set_password(PGSQL *pgsql, const char *username, const char *password)
+{
+	char command[BUFSIZE];
+
+	sformat(command, sizeof(command), "ALTER USER %s PASSWORD '%s'", username,
+			IS_EMPTY_STRING_BUFFER(password) ? "''" : password);
```

Comment:
I think we need to either use `PQescapeLiteral` as documented at https://www.postgresql.org/docs/current/libpq-exec.html#LIBPQ-EXEC-ESCAPE-STRING ; or even better use an actual SQL parameter to the command here. Because `ALTER USER` is a DDL, we might have to do like we do in `pgsql_create_user` to escape the identifiers and the literals. Please copy what we do there.